File: core/src/test/java/javabot/operations/SeenOperationTest.java
package javabot.operations;

import com.antwerkz.sofia.Sofia;
import javabot.BaseMessagingTest;
import org.testng.annotations.Test;

public class SeenOperationTest extends BaseMessagingTest {
  @Test
  public void seen() {
    testMessage("~seen jimmyjimjim", Sofia.seenUnknown(getTestUser().getNick(), "jimmyjimjim"));
  }
}

File: src/test/kotlin/javabot/operations/SeenOperationTest.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import javabot.BaseTest
import javax.inject.Inject
import org.testng.Assert
import org.testng.annotations.Test

class SeenOperationTest : BaseTest() {
    @Inject private lateinit var operation: SeenOperation

    @Test
    fun seen() {
        var response = operation.handleMessage(message("~seen jimmyjimjim"))
        Assert.assertEquals(response[0].value, Sofia.seenUnknown(TEST_USER.nick, "jimmyjimjim"))
    }
}
